farm/contract/lib.rs

```diff
@@ -315,10 +315,15 @@ mod farm {
             let undistributed_balance = if let Some(token_index) =
                 self.reward_tokens.iter().position(|&t| t == token)
             {
+                // After withdrawing all of the rewards, the reward rate should be zero.
+                self.farm_reward_rates[token_index] = 0;
                 total_balance.saturating_sub(self.farm_distributed_unclaimed_rewards[token_index])
             } else {
                 total_balance
             };
+            if self.farm_reward_rates.iter().all(|rr| *rr == 0) {
+                return Err(FarmError::AllRewardRatesZero);
+            }
```

Comment:
Why would we want to fail the call on all reward rates being 0?

That seems to be a valid scenario: owner wants to stop the farm and retrieve all the undistributed rewards. 
With this change it would not be possible, as withdrawing the last reward would always fail.

Reply:
If we allow for that, owner will be able to shut down an already-scheduled farm. Imagine creating a farm with lots of rewards, that start in 2 days. Farmers flock and put their LPs here. Now, the farmer withdraws all the rewards - the farm should be closed in such case, to not be started w/o any rewards. And farmers are left in an inactive farm.